From 75bacd3835eb83ba1e7c8e585cfe15cd67321109 Mon Sep 17 00:00:00 2001
From: Helmut Buchsbaum <helmut.buchsbaum@opensource.tttech-industrial.com>
Date: Thu, 2 Apr 2020 17:49:11 +0200
Subject: misc/efi-stub: Fix cast warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix the following warning;

boot.c: In function ‘switch_to_guest_mode’:
boot.c:310:24: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
  mbi->mi_loader_name = (UINT32)uefi_boot_loader_name;

Signed-off-by: Helmut Buchsbaum <helmut.buchsbaum@opensource.tttech-industrial.com>
---
 misc/efi-stub/boot.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/misc/efi-stub/boot.c b/misc/efi-stub/boot.c
index 529775db..d7089ac6 100644
--- a/misc/efi-stub/boot.c
+++ b/misc/efi-stub/boot.c
@@ -307,7 +307,7 @@ switch_to_guest_mode(EFI_HANDLE image, EFI_PHYSICAL_ADDRESS hv_hpa)
 	 * The host physical start address of boot loader name is stored in multiboot header.
 	 */
 	mbi->mi_flags |= MULTIBOOT_INFO_HAS_LOADER_NAME;
-	mbi->mi_loader_name = (UINT32)uefi_boot_loader_name;
+	mbi->mi_loader_name = (UINT64)uefi_boot_loader_name;
 
 	asm volatile ("pushf\n\t"
 		      "pop %0\n\t"
-- 
2.20.1

